src/components/application_manager/test/commands/mobile/on_system_request_notification_test.cc

```diff
@@ -78,7 +81,62 @@ class OnSystemRequestNotificationTest
   MockMessageHelper& message_helper_;
 };
 
-TEST_F(OnSystemRequestNotificationTest, Run_ProprietaryType_SUCCESS) {
+#ifdef EXTENDED_POLICY
+bool CheckHeader(MessageSharedPtr msg) {
+  struct DataValidator {
+    bool& is_valid_;
+    const utils::json::JsonValue& json_data_;
+
+    DataValidator(bool& is_valid, const utils::json::JsonValue& data)
+        : is_valid_(is_valid = true), json_data_(data) {}
+
+    void operator()(const std::string& str) {
+      if (!json_data_.HasMember(str)) {
+        is_valid_ = false;
+      }
+    }
+  };
+
+  std::string data_string(reinterpret_cast<const char*>(
+      &(*msg)[strings::params][strings::binary_data].asBinary()[0]));
+
+  utils::json::JsonValue json_data(
+      utils::json::JsonValue::Parse(data_string).first);
+
+  const std::string http_request = "HTTPRequest";
+  if (!json_data.HasMember(http_request))
+    return (false);
+  json_data = json_data.Get(http_request, json_data);
+
+  const std::string headers = "headers";
+  if (!json_data.HasMember(headers))
+    return (false);
+  json_data = json_data.Get(headers, json_data);
+
+  std::set<std::string> required_members();
+  required_members.insert("ContentType");
+  required_members.insert("ConnectTimeout");
+  required_members.insert("DoOutput");
+  required_members.insert("Content-Length");
+  required_members.insert("DoInput");
+  required_members.insert("UseCaches");
+  required_members.insert("RequestMethod");
+  required_members.insert("ReadTimeout");
+  required_members.insert("InstanceFollowRedirects");
+  required_members.insert("charset");
+
+  bool is_valid;
```

Comment:
@LevchenkoS I would prefere to handle valid result inside DataValidator to avoid boolean variable that change inside DataValidator. It is better to create DataValidator, and and after for_each check if it is valid.